### PR TITLE
fix Tensorflow and Pytorch Layer Tests fail on macOS

### PR DESCRIPTION
**Pull request by liubo-intel**
### Details:
 - *Tensorflow and Pytorch Layer Tests fail on macOS, because CPU plugin will use FP16 execution precision for some arm platforms*

### Tickets:
 - *122182*

### CODE REVIEW

File: tests/layer_tests/common/layer_test_class.py [hunk 1]
```diff
@@ -93,9 +93,9 @@ def _test(self, framework_model, ref_net, ie_device, precision, ir_version, temp
         #     assert flag, '\n'.join(resp)
 
         config = None
-        # GPU default execution precision is FP16, so if we want to check FP32 inference
+        # GPU and some CPU arm platforms default execution precision is FP16, so if we want to check FP32 inference
         # we need to set explicit precision hint
-        if ie_device == 'GPU' and precision == 'FP32':
+        if precision == 'FP32':
```
Comment (rkazants) [inline tests/layer_tests/common/layer_test_class.py:98]: I think we test the default scenario in the layer test that is more important.

Comment (rkazants) [reply tests/layer_tests/common/layer_test_class.py:98]: Is it possible to figure out why the accuracy is degraded for the default scenario on these plugins? This fix can just hide the issue in the plugins.

Comment (liubo-intel) [reply tests/layer_tests/common/layer_test_class.py:98]: hi, @rkazants :  because CPU Plugin use fp16 inference precision by default for some specific arm platform(macOS) cases. for these cases, Nodes will be converted to fp16 precision during runtime. detail information can be found in the ticket-122182

Comment (rkazants) [reply tests/layer_tests/common/layer_test_class.py:98]: We should test the default scenario and no need to change inference precision. If you need to top up threshold, it is possible to configure in the test.

**Review by rkazants — CHANGES_REQUESTED** [review]
Please test the default inference precision and set custom accuracy threshold if needed

File: tests/layer_tests/common/layer_test_class.py (hunk 1 above)
Comment (mvafin) [reply tests/layer_tests/common/layer_test_class.py:98]: Agree with Roman, it is better to test default scenario. GPU+FP16 and ARM+FP16. It will uncover problems in FP16 inference which we definitely have currently

**Comment by mvafin** [comment]
Pytorch layer tests are not run in precommit on ARM and GPU, please provide your local logs that tests pass with your change

File: tests/layer_tests/common/layer_test_class.py (hunk 1 above)
Comment (liubo-intel) [reply tests/layer_tests/common/layer_test_class.py:98]: yes, agree that we can try to top up threshold for these kinds of cases to solve this issue. 
But does it seems functional mismatch if the layer tests really run in fp16 precision(e.g.  we don't set {''INFERENCE_PRECISION_HINT': 'f32'} for some ARM platforms) while  layer tests script [set precision to fp32](https://github.com/openvinotoolkit/openvino/blob/bd57987c36f71a0f47e1dc3755a6bb7a706c12dd/tests/layer_tests/README.md?plain=1#L28)? I mean dose this [precision](https://github.com/openvinotoolkit/openvino/blob/bd57987c36f71a0f47e1dc3755a6bb7a706c12dd/tests/layer_tests/README.md?plain=1#L28) stands for the target test precision?

Comment (rkazants) [reply tests/layer_tests/common/layer_test_class.py:98]: I think it should be a question to plugins, why we have some deconvergence between plugins so that different precision is used by default.
Please test default scenario and set separate thresholds for ARM. Otherwise, some issue can e hidden in the plugin

Comment (liubo-intel) [reply tests/layer_tests/common/layer_test_class.py]: have changed to keep test default scenario and set separate thresholds(former fp16 threshold) for ARM CNN model cases to align with CPU Plugin

**Comment by liubo-intel** [comment]
hi, @rkazants : if you still have concern about this pr' change or decide to fix this issue from layer_test architecture adjustment and optimization, maybe I should close this pr currently?

File: tests/layer_tests/common/layer_utils.py [hunk 2]
```diff
@@ -99,3 +99,13 @@ def get_inputs_info(self, precision) -> dict:
...
+
+    def cnn_model_type_check(self):
+        core = Core()
+        net = core.read_model(self.model, self.weights)
+        arm_cnn_types = ["Convolution", "ConvolutionBackpropData"]
+        for op in net.get_ops():
+            op_type = op.get_type_info().name
+            if op_type in arm_cnn_types:
+                return True
+        return False
```
Comment (rkazants) [inline tests/layer_tests/common/layer_utils.py:111]: let us instead of this, point threshold in the test itself using `kwargs["custom_eps"]`. See examples in other tests

File: tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py [hunk 3]
```diff
@@ -174,7 +175,28 @@ def numpy_to_torch_recursively(x):
...
+                        break
+            if precision == 'FP32':
+                if not arm_platform:
+                    fw_eps = custom_eps
+                elif arm_cnn_model:
+                    fw_eps = custom_eps
+                else:
+                    fw_eps = 5e-2
+            else:
+                fw_eps = 5e-2
```
Comment (rkazants) [inline tests/layer_tests/pytorch_tests/pytorch_layer_test_class.py:198]: let us instead of this, point threshold in the test itself using kwargs["custom_eps"]. See examples in other tests
